controllers/articles.controllers.js

```diff
@@ -1,19 +1,26 @@
 const { fetchArticleById, fetchAllArticles, fetchAllCommentsByArticleId, checkArticleExists, updateArticleVotes } = require("../models/articles.models")
+const { checkTopicExists } = require("../models/topics.models")
 
 function getArticleById(request, response, next) {
     const { article_id } = request.params
     fetchArticleById(article_id)
         .then((article) => {
             response.status(200)
-            .send({ article: article })
+            .send({ article })
         }).catch((error) => {
             next(error)
         })
 }
 
 function getArticles(request, response, next) {
-    fetchAllArticles()
-        .then((articles) => {
+    const { topic } = request.query
+    promises = [fetchAllArticles(topic)]
+    if (topic) {
+        promises.push(checkTopicExists(topic))
```

Comment:
We don't want to risk SQL injection by potentially running fetchAllArticles(topic) before/same time as checking the topic exists. I would therefore make sure we do checkTopicExists and then we can do fetchAllArticles

Promise.all does not guarantee the order in which the promises are fulfilled.